.github/workflows/build-packages.yml

```diff
@@ -117,42 +134,50 @@ jobs:
       - name: Upload artifact
         uses: actions/upload-artifact@v2
         with:
-          name: bob-deb
+          name: bob-deb-${{ matrix.bob_key_size }}
           path: bob_${{ env.VERSION }}_amd64.deb
 
-
   upload-deb:
     runs-on: ubuntu-latest
     needs: build-deb
     environment: prod
+    strategy:
+      matrix:
+        bob_key_size: [8, 16]
+        include:
+          - bob_key_size: 16
+            asset_name_postfix: _key16
```

Comment:
You can use conditional step that generates this name and save it to environment variable only when `bob_key_size != 8`